internal/server/config.go

```diff
@@ -205,8 +209,8 @@ func (kp *KeyProvider) PrepareForDecode(data interface{}) error {
 }
 
 type SecretProvider struct {
-	Kind   string      `mapstructure:"kind"`
-	Name   string      `mapstructure:"name"`
+	Kind   string      `config:"kind"`
```

Comment:
We shouldn't need these aliases anymore in most cases. I haven't removed all the `mapstructure` tags, but we will be able to after this.

This case is special because this field is read by `SecretProvider.PrepareForDecode`. In that method we don't look for the key case-insensitive, so it has to be lowercase here.  

We could fix that by allowing `PrepareForDecode` to case insensitive match `kind`, but I want to keep that change separate, since this is already a pretty large PR.